### PR TITLE
Improve display of validation errors for processing instructions and duplicate elements

### PR DESCRIPTION
**Pull request by westonruter**
## Summary

In looking at https://github.com/ampproject/amp-wp/issues/3409#issuecomment-547213190 I noticed that validation errors for processing instructions aren't being displayed properly. This fixes that.

Given a Custom HTML block that looks like:

image

The changes look as follows:

Before | After
-------|------
image | image
image | image
Screen Shot 2019-10-28 at 22 45 39 | Screen Shot 2019-10-28 at 22 45 51

This also fixes the presentation of `duplicate_element` errors, which was noticed when the AMP-to-AMP linking plugin was active after #3627 was merged.

Before | After
-------|-----
Screen Shot 2019-10-29 at 11 25 38 | Screen Shot 2019-10-29 at 11 25 31

See #3409. Amends #3544.

## Checklist

- [ ] My pull request is addressing an [open issue](https://github.com/ampproject/amp-wp/contributing/project-management.md#life-of-an-issue) (please create one otherwise).
- [ ] My code is tested and passes existing [tests](https://github.com/ampproject/amp-wp/contributing/engineering.md#tests).
- [ ] My code follows the [Engineering Guidelines](https://github.com/ampproject/amp-wp/contributing/engineering.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

**Comment by kienstra** [comment]
Hi @westonruter,
Thanks for requesting my review, I'll start review this in a few hours is that's alright.

File: includes/validation/class-amp-validation-error-taxonomy.php [hunk 1]
```diff
@@ -1762,7 +1769,12 @@ public static function get_reader_friendly_error_type_text( $error_type ) {
 	 * @return string The label.
 	 */
 	public static function get_details_summary_label( $validation_error ) {
-		if ( self::INVALID_ATTRIBUTE_CODE === $validation_error['code'] || self::INVALID_ELEMENT_CODE === $validation_error['code'] ) {
+		if (
+			self::INVALID_ATTRIBUTE_CODE === $validation_error['code'] ||
+			self::INVALID_ELEMENT_CODE === $validation_error['code'] ||
+			'invalid_processing_instruction' === $validation_error['code'] ||
```
Comment (kienstra) [inline includes/validation/class-amp-validation-error-taxonomy.php]: Maybe `'invalid_processing_instruction'` could be in a class constant like `self::INVALID_PROCESSING_INSTRUCTION_CODE`, similar to some other codes above.

**Comment by kienstra** [comment]
**Approved**

Hi @westonruter,
Good idea to add the `'invalid_processing_instruction'` code.

It appeared as expected:
new-errror-code-here

error-appears-markup

Also, other error codes continued to display as expected, like the invalid attribute code for `onclick`:

attribute-onclick-here

File: includes/validation/class-amp-validation-error-taxonomy.php (hunk 1 above)
Comment (westonruter) [reply includes/validation/class-amp-validation-error-taxonomy.php]: Yes, I thought about that, but I want to wait to add more constants until we decide what to do with aligning error codes with the AMP Validator (#1420).

Comment (kienstra) [reply includes/validation/class-amp-validation-error-taxonomy.php]: Sure, that sounds good.

**Pull request merged**